core/src/main/java/org/elasticsearch/indices/ttl/IndicesTTLService.java

```diff
@@ -340,7 +340,7 @@ public void await() {
             try {
                 condition.await(timeout.millis(), TimeUnit.MILLISECONDS);
             } catch (InterruptedException e) {
-                Thread.interrupted();
+                Thread.currentThread().interrupt();
```

Comment:
This is used by an internal thread where we control the run loop so the interruption here is also fine. However, I am not sure why we create threads ourselves outside of pools. Btw, I think this thread's implementation could be simplified significantly if we stick to Java's standard methods, i.e. interruption :) (but I would not do this in the scope of this PR).